src/components/EditExercise.tsx

```diff
@@ -0,0 +1,60 @@
+
+import ExerciseForm from "./ExerciseForm";
+import { useState , useEffect} from "react";
+import exerciseService from "../services/exercises";
+import { useNavigate, useParams } from "react-router-dom";
+
+const EditExercise = () => {
+     
+  const { id } = useParams<{ id?: string }>();
+  
+  console.log(id)
+  const [title, setTitle] = useState("");
+  const [complexity, setComplexity] = useState("");
+  const [textExercise, setTextExercise] = useState("");
+  const navigate = useNavigate();
+
+  // Effect to fetch exercise data when the component mounts
+  useEffect(() => {
+    exerciseService
+      .getByID(id!)
```

Comment:
Make sure id exists with some sort of if (id), otherwise it may give an error
Using the ! in typescript, you are removing the null check safeguard

Reply:
Done